visualizations/frontend/other/TPdfTable.vue

```diff
@@ -0,0 +1,149 @@
+<template>
+  <div class="relative overflow-auto">
+    <t-loading-spinner v-if="loading" position="relative" />
+    <slot name="header" :loading="loading"></slot>
+    <div :class="containerClass">
+      <table class="w-full table-fixed">
+        <thead>
+          <th v-for="column in columns" :key="column" :class="columnsClass">
+            <slot :name="'col-' + column" :column="column">
+              {{ column }}
+            </slot>
+          </th>
+        </thead>
+        <tbody>
+          <tr v-for="(row, index) in rows" :key="'row-' + index">
+            <td
+              v-for="column in columns"
+              :key="column"
+              class="break-all"
+              :class="rowsClass"
+            >
+              <slot :name="'row-' + column" :row="row">
+                {{ row[column] ? row[column].rendered : "" }}
+              </slot>
+            </td>
+          </tr>
+        </tbody>
+        <div v-if="!loading && !rows.length" class="text-center">No Data</div>
+      </table>
+    </div>
+    <slot
+      name="footer"
+      :rows-limit="rowsLimit"
+      :rows-total="rowsTotal"
+      :loading="loading"
+    >
+      <div class="pt-4 text-right">
+        Showing {{ rowsLimit }} of {{ rowsTotal }}
+      </div>
+    </slot>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    hiddenColumns: {
+      type: Array,
+      default: () => [],
+    },
+    dynamicColumns: {
+      type: Array,
+      default: () => [],
+    },
+    customColumns: {
+      type: Array,
+      default: () => [],
+    },
+    rowsLimit: {
+      type: Number,
+      default: 5,
+    },
+    containerClass: {
+      type: String,
+      default: "",
+    },
+    columnsClass: {
+      type: String,
+      default: "",
+    },
+    rowsClass: {
+      type: String,
+      default: "",
+    },
```

Comment:
Can you tell me why you need these?
Would it also be acceptable to enforce a certain table style?

Reply:
Table keeps on evolving on every page inside multiple figmas I have seen so far, so I thought it's better to just allow full customization from outside without having to change the table from inside or force a specific design by default.

Reply:
We recently had an initiative to consolidate the design of the table so we don't need to maintain so many styling variations and can maintain design consistency across reports. Here are the related PRs:
* https://github.com/topcoat-data/topcoat-public/pull/240
* https://github.com/topcoat-data/topcoat-public/pull/239

While I'm not opposed to providing these props if there is a strong need, I think they pose the problem of a table the diverges with designs by accident or on purpose. In addition, you need to repeat style props at every usage point.

Reply:
It's okay. We can always address a refactor if the maintenance overhead arises.

Reply:
Got it, thanks.